src/13_integrate_BBBP.py

```diff
@@ -19,55 +21,41 @@
     .config("spark.network.timeout","10000000ms") \
     .getOrCreate()
 
-stg = os.makedirs("staging/BBBP", exist_ok=True)
+os.makedirs("staging/BBBP", exist_ok=True)
 data = biobricks.assets("MoleculeNet")
 
-# BUILD COMPOUNDS =========================================================
+# Build compounds
 bbbp_raw = spark.read.parquet(data.BBBP_parquet)
-
-# Filter out rows with null SMILES values
 bbbp_filtered = bbbp_raw.filter(F.col('smiles').isNotNull())
 bbbp_with_sid = bbbp_filtered.withColumn('sid', F.monotonically_increasing_id().cast('string'))
 
-# Combine metadata into the data column
-subjson = bbbp_with_sid.select(
+# Convert SMILES to InChI
+smiles_to_inchi_udf = get_smiles_to_inchi_udf()
+bbbp_with_inchi = bbbp_with_sid.withColumn("inchi", smiles_to_inchi_udf(F.col("smiles"))).cache()
+
+# Substances table
+subjson = bbbp_with_inchi.select(
     "sid", 
-    F.to_json(
-        F.struct(
-            F.col("num"),
-            F.col("name"),
-            F.col("p_np"),
-            F.col("smiles")
-        )
-    ).alias("data")
+    F.to_json(F.struct("num", "name", "smiles", "inchi")).alias("data")
 )
-
-# Write the substances table to Parquet
 subjson.write.mode("overwrite").parquet("staging/BBBP/substances.parquet")
 
-# WRITE PROPERTIES =====================================================
-# Define the JSON data for properties
+# Properties table
 data_json = json.dumps({
-    'p_np = 1': 'permeable',
-    'p_np = 0': 'non-permeable'
+    'property': 'p_np',
```

Comment:
the property metadata should let somebody read the property and understand what they are looking at. p_np isn't descriptive enough.